dask/layers.py

```diff
@@ -43,9 +44,21 @@ class CreateArrayDeps(BlockwiseIODeps):
 
     def __init__(self, chunks: tuple):
         self.chunks = chunks
+        self.shape = tuple(sum(c) for c in chunks)
+        self.starts = [cached_cumsum(c, initial_zero=True) for c in chunks]
+        self.num_chunks = tuple(len(c) for c in chunks)
 
     def __getitem__(self, idx: tuple):
-        return tuple(chunk[i] for i, chunk in zip(idx, self.chunks))
+        chunk_shape = tuple(chunk[i] for i, chunk in zip(idx, self.chunks))
+        array_location = tuple(
+            (start[i], start[i + 1]) for i, start in zip(idx, self.starts)
+        )
+        return {
+            "shape": self.shape,
+            "num-chunks": self.num_chunks,
+            "array-location": array_location,
+            "chunk-shape": chunk_shape,
+        }
```

Comment:
Perhaps we should make it possible to avoid unnecessary logic for "simple" cases like zeros/full/ones?

Reply:
This `block_info` dict is modeled after that in `Array.map_blocks` (though it's not identical, since that can include dtypes). My thinking was that, since they are similar operations, the API should be similar, and similar data should be provided to the tasks.

That being said, it's true that in simpler cases there is some unnecessary logic here when constructing the block info. We could revert these changes to keep `CreateArrayDeps` simple, and then make a new blockwise layer for things like `from_array`. So to me it seems the choice is between having a larger number of more specific layer classes, or a smaller number of more general ones like this one.

Reply:
Or `__init__` could take something like `block_info=True` to indicate that it should provide more than just the chunk shape. I generally like to have more predictable outputs for functions, but can be talked out of it :)

Reply:
>  I generally like to have more predictable outputs for functions, but can be talked out of it :)

Makes sense - I do prefer consistency (as you already have it), so I'm not really motivated to talk you out of this unless we discover a need for performance optimizations.